artifactory/commands/transferfiles/utils.go

```diff
@@ -151,7 +153,12 @@ func pollUploads(phaseBase *phaseBase, srcUpService *srcUserPluginService, uploa
 						continue
 					}
 				}
-				curTokensBatch.UuidTokens = removeTokenFromBatch(curTokensBatch.UuidTokens, chunk.UuidToken)
+				err := awaitingStatusChunksSet.Remove(chunk.UuidToken)
+				if err != nil {
+					log.Error("removal from upload status set issue occurred: " + err.Error())
+					continue
+				}
+				curTokensBatch.ChunksToDelete = append(curTokensBatch.ChunksToDelete, chunk.UuidToken)
```

Comment:
Let's add a comment here explaining why we're doing this. Let's make it clear for the random reader what `ChunksToDelete` actually means and why it is needed.

Reply:
done